spikes/oms-log-analytics/README.md

```diff
@@ -4,59 +4,60 @@ Prometheus is a popular open source metric monitoring solution and is a part of
-   1. Check if `omsagent` Pod is running: `kubectl get pods -A | grep omsagent`
-   2. If `omsagent` is running then only apply the config file: `kubectl apply -f 3-container-azm-ms-agentconfig.yaml`
-   3. Skip to Step 6
-3. To use OMS Agent Daemonset and deployment:
-   1. Create the log analytics secrets with kubectl:
+For non-AKS clusters please see [non-aks/README.md](non-aks/README.md).
 
-    ```bash
+For installing on AKS clusters, we recommend enabling Azure Monitor for Containers through your Infrastructure as Code solution.
+Please see, [Enable Container Insights](https://docs.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-onboard) for instructions on enabling Azure Monitor for Containers for your cluster.
 
-    kubectl create secret generic omsagent-secret -n kube-system \
-                --from-literal=KEY=$(az monitor log-analytics workspace get-shared-keys \
-                -g LOG_ANALYTICS_GROUP -n LOG_ANALYTICS_NAME --query primarySharedKey -o tsv) \
-                --from-literal=WSID=$(az monitor log-analytics workspace show \
-                -g ngsa-fb-test-test-rg -n ngsa-fb-test-log --query customerId -o tsv)
-    # If you don't have az cli access, replace the $(az monitor...) portion with actual values
-    
-    ```
+For more detail on what is created, please see the YAML files in `./non-aks`.
 
-   2. Check secrets created, run `kubectl get secrets omsagent-secret -n kube-system -o jsonpath='{.data}'`
-   3. Properly annotate target pods so that Prometheus can select and scrape the pod metrics. For details on proper annotations for OMS agent see [this link](https://docs.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-prometheus-integration)
+Once Azure Monitor for Containers has been deployed, you need to apply a config map to enable application log and metric collection.
+An example config map is provided in [container-azm-ms-agentconfig.yaml](./container-azm-ms-agentconfig.yaml).
+This can be applied through `kubectl` or a gitops solution like Flux.
+For `kubectl`:
 
-    ```bash
-    
-    # Make sure --prometheus argument has been passed to ngsa-memory or ngsa-cosmos
-    # Otherwise, ngsa won't output Prometheus data to `/metrics` path
+```bash
+kubectl apply -f container-azm-ms-agentconfig.yaml
+```
 
-    # Annotate the pod instance "ngsa-memory"
-    kubectl annotate pod ngsa-memory prometheus.io/scrape=true --overwrite prometheus.io/path='/metrics' prometheus.io/port=8080 prometheus.io/scheme=http interval=30s --overwrite
+The provided config map enables:
 
-    # Check ngsa-memory has the correct annotations
-    kubectl get pod ngsa-memory -o jsonpath='{.metadata.annotations}' | jq
+* Collection of application logs
+* Collection of application metrics
+  * Note: With this Config Map, Azure Monitor For Containers relies on pod annotations to point to the port and endpoint for metric collection.
+  * Pods should be annotated with the following labels:
```

Comment:
To clarify that the values of these labels are not hard-coded, this line might be added

"`path`, `port`, `scheme` and `interval` should be specific to your application"